### ISSUE TITLE
Predict failures on Big Endian system

### ISSUE THREAD

**Issue by tdoublep** [dmlc/xgboost#7037]
I'm running XGBoost on Linux on IBM Z (s390x architecture, a big endian system) and something pretty major seems to have changed between version 1.3.3 and 1.4.0. In particular, the following very simple test case seems to fail:

```python
import sklearn
import xgboost
import numpy as np

print("sklearn version: ", sklearn.__version__)
print("xgboost version: ", xgboost.__version__)
print("numpy version:   ", np.__version__)

from sklearn.datasets import make_classification
X, y = make_classification(n_samples=100, n_features=10, random_state=42)

m = xgboost.XGBClassifier(use_label_encoder=False, eval_metric='logloss')
m.fit(X, y)
m.predict(X)
```
With `xgboost==1.3.3`, everything works:
```bash
sklearn version:  0.24.1
xgboost version:  1.3.3
numpy version:    1.19.2
```
With `xgboost==1.4.0`:
```
sklearn version:  0.24.1
xgboost version:  1.4.0
numpy version:    1.19.2
Traceback (most recent call last):
  File "test.py", line 17, in <module>
    m.predict(X)
  File "/app/xgboost/python-package/xgboost/sklearn.py", line 1215, in predict
    iteration_range=iteration_range,
  File "/app/xgboost/python-package/xgboost/sklearn.py", line 826, in predict
    validate_features=validate_features,
  File "/app/xgboost/python-package/xgboost/core.py", line 1839, in inplace_predict
    f"Feature shape mismatch, expected: {self.num_features()}, "
ValueError: Feature shape mismatch, expected: 0, got 100
```
With latest commit on master (`2567404ab601149c52b66c3a02d85e9571a84447`), I see a different error still:
```
sklearn version:  0.24.1
xgboost version:  1.5.0-dev
numpy version:    1.19.2
Traceback (most recent call last):
  File "test.py", line 13, in <module>
    m.fit(X, y)
  File "/app/xgboost/python-package/xgboost/core.py", line 450, in inner_f
    return f(**kwargs)
  File "/app/xgboost/python-package/xgboost/sklearn.py", line 1222, in fit
    label_transform=label_transform,
  File "/app/xgboost/python-package/xgboost/sklearn.py", line 283, in _wrap_evaluation_matrices
    enable_categorical=enable_categorical,
  File "/app/xgboost/python-package/xgboost/sklearn.py", line 1220, in <lambda>
    create_dmatrix=lambda **kwargs: DMatrix(nthread=self.n_jobs, **kwargs),
  File "/app/xgboost/python-package/xgboost/core.py", line 450, in inner_f
    return f(**kwargs)
  File "/app/xgboost/python-package/xgboost/core.py", line 561, in __init__
    enable_categorical=enable_categorical,
  File "/app/xgboost/python-package/xgboost/data.py", line 544, in dispatch_data_backend
    feature_types)
  File "/app/xgboost/python-package/xgboost/data.py", line 148, in _from_numpy_array
    ctypes.byref(handle),
  File "/app/xgboost/python-package/xgboost/core.py", line 213, in _check_call
    raise XGBoostError(py_str(_LIB.XGBGetLastError()))
xgboost.core.XGBoostError: [13:13:45] /app/xgboost/src/c_api/../data/array_interface.h:142: Check failed: typestr.front() != '>' (> vs. >) : Big endian is not supported.
Stack trace:
  [bt] (0) /app/xgboost/python-package/xgboost/../../lib/libxgboost.so(+0xd5c10) [0x3ff9d955c10]
  [bt] (1) /app/xgboost/python-package/xgboost/../../lib/libxgboost.so(xgboost::ArrayInterfaceHandler::Validate(std::map<std::string, xgboost::Json, std::less<std::string>, std::allocator<std::pair<std::string const, xgboost::Json> > > const&)+0x52c) [0x3ff9d970dc4]
  [bt] (2) /app/xgboost/python-package/xgboost/../../lib/libxgboost.so(xgboost::ArrayInterface::Initialize(std::map<std::string, xgboost::Json, std::less<std::string>, std::allocator<std::pair<std::string const, xgboost::Json> > > const&, bool)+0x4a) [0x3ff9d974f2a]
  [bt] (3) /app/xgboost/python-package/xgboost/../../lib/libxgboost.so(XGDMatrixCreateFromDense+0xe0) [0x3ff9d95d9d0]
  [bt] (4) /lib64/libffi.so.6(ffi_call_SYSV+0x98) [0x3ffa4585860]
  [bt] (5) /lib64/libffi.so.6(ffi_call+0x76) [0x3ffa4585f56]
  [bt] (6) /opt/python/cp37-cp37m/lib/python3.7/lib-dynload/_ctypes.cpython-37m-s390x-linux-gnu.so(_ctypes_callproc+0x2ee) [0x3ffa469369e]
  [bt] (7) /opt/python/cp37-cp37m/lib/python3.7/lib-dynload/_ctypes.cpython-37m-s390x-linux-gnu.so(+0x9eec) [0x3ffa4689eec]
  [bt] (8) /opt/python/cp37-cp37m/bin/python(_PyObject_FastCallKeywords+0x114) [0x103e93c]
```
Is this the expected behaviour? The last error suggests that Big Endian support has been dropped - Is this correct?

**Comment by trivialfis** [comment]
I will try to add it back.  We dropped support for some of of datasets due to using array interface.

**Comment by tdoublep** [comment]
If I can help in any way (e.g. by trying stuff out on s390x system) please let me know.

**Comment by trivialfis** [comment]
Just restored travis test on s390x https://github.com/dmlc/xgboost/pull/7038 .  Basic tests are passing.

**Comment by tdoublep** [comment]
Confirming that this issue is fixed by #7038

**Comment by trivialfis** [comment]
Excellent!  @hcho3 Could you please take a look at the PR?
